stacks_queues/queue.py

```diff
@@ -17,45 +17,65 @@ def __init__(self):
         self.size = 0
       
 
+    # methods to add to the Queue class
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
+        
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.size == 0:
+            raise QueueEmptyException
+        element = self.store[self.front]
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+        return element
+
+
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.front == -1:
+            return None
```

Comment:
Because you never check in `dequeue` whether dequeuing an element makes the Queue empty and, if the Queue does become empty, reset `front` to point at index -1, it may not always be true that the front pointer is at index -1 when the queue is empty. Try using the `empty` method you implement instead.
```suggestion
        if self.empty():
            return None
```